src/Http/Http.Extensions/src/RequestDelegateFactoryOptions.cs

```diff
@@ -36,4 +36,14 @@ public sealed class RequestDelegateFactoryOptions
     /// The list of filters that must run in the pipeline for a given route handler.
     /// </summary>
     public IReadOnlyList<Func<RouteHandlerContext, RouteHandlerFilterDelegate, RouteHandlerFilterDelegate>>? RouteHandlerFilterFactories { get; init; }
+
+    /// <summary>
+    /// The default endpoint metadata to add as part of the creation of the <see cref="RequestDelegateResult.RequestDelegate"/>.
+    /// </summary>
+    /// <remarks>
+    /// This metadata will be included in <see cref="RequestDelegateResult.EndpointMetadata" /> after any metadata inferred during creation of the
+    /// <see cref="RequestDelegateResult.RequestDelegate"/> but before any metadata provided by types in the delegate signature that implement
+    /// <see cref="IEndpointMetadataProvider" /> or <see cref="IEndpointParameterMetadataProvider" />.
```

Comment:
Why isn't this the following?

```suggestion
    /// This metadata will be included in <see cref="RequestDelegateResult.EndpointMetadata" /> before any metadata inferred during creation of the
    /// <see cref="RequestDelegateResult.RequestDelegate"/> and before any metadata provided by types in the delegate signature that implement
    /// <see cref="IEndpointMetadataProvider" /> or <see cref="IEndpointParameterMetadataProvider" />.
```

I see that it's currently implemented as described, but wouldn't it be more logical to put all the metadata inferred by RDF and inferred by MetadataProviders together after the `DefaultEndpointMetadata`.

Reply:
Because earlier metadata items have precedence when building the `ApiDescription`. The metadata provided by `IEndpointMetadataProvider` types can manually override that precedence if they wish as they have access to the metadata as a list (i.e. that can change the order), but I had assumed we would want the metadata inferred from things like attributes to have higher precedence by default than metadata added by the return type itself, given that the attribute is manually added to the method by the developer.

Reply:
So I've hit an interesting wrinkle with this that either proves you right @halter73 (and my understanding wrong) or represents a mismatch between what the routing system considers the most specific metadata (the last one of a given type in the list), and what our projection into `ApiExplorer` considers the most specific metadata (the first one of a given type in the list).

Reply:
To add clarity here, the metadata provided by the caller via `RequestDelegateFactoryOptions.AdditionalEndpointMetadata` includes custom attributes on the route handler delegate and should be **more** specific than any metadata inferred by RDF or added by RDF via types implementing `IEndpointMetadataProvider`/`IEndpointParameterMetadataProvider`. The most specific metadata item of a specific type is the **last** one in the metadata list, therefore the additional metadata should be added after anything inferred.

This enables the developer to ultimately override any metadata automatically added via an attribute on the route handler method or by calling `WithMetadata()` on the builder.